src/Repository.js

```diff
@@ -43,7 +44,34 @@ export default class Repository {
     return this.lernaJson && this.lernaJson.bootstrapConfig || {};
   }
 
+  get packages() {
+    if (!this._packages) {
+      this.buildPackageGraph();
+    }
+    return this._packages;
+  }
+
+  get filteredPackages() {
+    if (!this._filteredPackages) {
+      this.buildPackageGraph();
+    }
+    return this._filteredPackages;
+  }
+
+  get packageGraph() {
+    if (!this._packageGraph) {
+      this.buildPackageGraph();
```

Comment:
Cool, ended up doing exactly what you said and moved the responsibility in PackageUtilities and it only gets called in one place still (Command.js).

Much better!